pkg/operator/ceph/csi/secrets.go

```diff
@@ -44,7 +53,25 @@ const (
 	CsiCephFSProvisionerSecret          = "rook-csi-cephfs-provisioner"
 )
 
-func createCSIKeyringRBDNode(s *keyring.SecretStore) (string, error) {
+func createCSIKeyringRBDNode(context *clusterd.Context, clusterInfo *client.ClusterInfo, clusterSpec *cephv1.ClusterSpec, s *keyring.SecretStore, keepPriorCount uint32) (string, error) {
+	shouldRotate, allKeyWithBaseName, err := getCsiKeyRotationInfo(context, clusterInfo, clusterSpec, csiKeyringRBDNodeUsername)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to check if key rotation is supported for CSI key %s", csiKeyringRBDNodeUsername)
+	}
+
+	if shouldRotate {
+		key, err := s.GenerateKey(generateCsiUserIdWithGenerationSuffix(csiKeyringRBDNodeUsername, keepPriorCount), cephCSIKeyringRBDNodeCaps())
+		if err != nil {
+			return "", errors.Wrapf(err, "failed to check if keys should be rotated for CSI key %s", csiKeyringRBDNodeUsername)
+		}
+
+		err = deleteOldKeyGen(context, clusterInfo, allKeyWithBaseName, int(keepPriorCount))
```

Comment:
I think we need to call `deleteOldKeyGen` regardless of `shouldRotate`. Otherwise, the case where the user reduces `keepPriorKeyCount` from `1` to `0` wouldn't happen when no rotation is indicated.

With the code as-is, this would have to be changed in 4 places. I think this kind of change is why a helper method for this code section will be important for maintaining this code here and in the future

Reply:
moved the `deleteOldKeyGen` block out of  `if shouldRotate` block